### ISSUE TITLE
Any chance this can be published as a library so we can use it in other obsidian plugins?

### ISSUE THREAD

**Issue by Odaimoko** [pjeby/tag-wrangler#132]
I am maintaining [Tag Project](https://github.com/Odaimoko/tag-project) which utilizes tags to manage tasks.
Tag Wrangler can be extremely useful when renaming tags. If a library exists it would definitely help a lot for other devs.

**Comment by pjeby** [comment]
In principle, I suppose it could be?  It's really quite a mess, as it was one of my first Obsidian plugins and I originally wrote in JavaScript and it's not in very good shape, organizationally speaking.  It might be easier to for me to add API(s) to the plugin so that other plugins can call it, than to extract a proper library.

**Comment by Odaimoko** [comment]
> In principle, I suppose it could be? It's really quite a mess, as it was one of my first Obsidian plugins and I originally wrote in JavaScript and it's not in very good shape, organizationally speaking. It might be easier to for me to add API(s) to the plugin so that other plugins can call it, than to extract a proper library.

Maybe Tag Wrangler can listen to some event (with parameters) and act accordingly.

For example, dataview will trigger `dataview:index-ready` event on `app.metadataCache` when dataview's index is fully loaded. [Here's](https://github.com/blacksmithgu/obsidian-dataview/blob/0c54ac0b3cc83e72ad40913b3229ff02923243d6/src/data-index/index.ts#L152) the code.

Conversely, for renaming, Tag Wrangler can listen to `tagwrangler:replaceTagRequest` event on some obsidian's `Events` object. If some other plugin triggers this event with `oldTag` and `newTag` (including hashtag symbol `#`), tag wrangle can do the work.

**Comment by pjeby** [comment]
No, that would not allow errors or results to be passed back properly - it should be async methods on the plugin object most likely.  Not sure when I will have time to actually do this, though.
